morpheus/_lib/src/messages/multi_response_probs.cpp

```diff
@@ -56,18 +56,18 @@ std::shared_ptr<MultiResponseProbsMessage> MultiResponseProbsMessageInterfacePro
     std::shared_ptr<MessageMeta> meta,
     cudf::size_type mess_offset,
     cudf::size_type mess_count,
-    std::shared_ptr<ResponseMemory> memory,
+    std::shared_ptr<ResponseMemoryProbs> memory,
     cudf::size_type offset,
     cudf::size_type count)
 {
     return std::make_shared<MultiResponseProbsMessage>(
         std::move(meta), mess_offset, mess_count, std::move(memory), offset, count);
 }
 
-std::shared_ptr<morpheus::ResponseMemory> MultiResponseProbsMessageInterfaceProxy::memory(
+std::shared_ptr<morpheus::ResponseMemoryProbs> MultiResponseProbsMessageInterfaceProxy::memory(
     MultiResponseProbsMessage &self)
 {
-    return self.memory;
+    return DCHECK_NOTNULL(std::dynamic_pointer_cast<morpheus::ResponseMemoryProbs>(self.memory));
```

Comment:
Can this be a `std::static_pointer_cast` instead? We can add a `DCHECK` beforehand that does the dynamic cast just to be sure.